composer/lib/dependabot/composer/helpers.rb

```diff
@@ -6,8 +6,13 @@
 module Dependabot
   module Composer
     module Helpers
+      V1 = "1"
+      V2 = "2"
+      # If we are updating a project with no lock file then the default should be the newest version
+      DEFAULT = V2
+
       # From composers json-schema: https://getcomposer.org/schema.json
-      COMPOSER_V2_NAME_REGEX = %r{^[a-z0-9]([_.-]?[a-z0-9]+)*/[a-z0-9](([_.]?|-{0,2})[a-z0-9]+)*$}
+      COMPOSER_V2_NAME_REGEX = %r{^[a-z0-9]([_.-]?[a-z0-9]++)*/[a-z0-9](([_.]?|-{0,2})[a-z0-9]++)*$}
```

Comment:
Updated quantifiers from + to ++ to prevent backtracking and improve performance, addressing potential ReDoS vulnerabilities flagged by CodeQL.

Resolved CodeQL: https://github.com/dependabot/dependabot-core/pull/10716#discussion_r1786919691